statistics/src/main/java/com/duckduckgo/app/statistics/AtbInitializer.kt

```diff
@@ -16,27 +16,41 @@
 
 package com.duckduckgo.app.statistics
 
-import com.duckduckgo.app.referral.AppInstallationReferrerStateListener
-import com.duckduckgo.app.referral.AppInstallationReferrerStateListener.Companion.MAX_REFERRER_WAIT_TIME_MS
 import com.duckduckgo.app.statistics.api.StatisticsUpdater
 import com.duckduckgo.app.statistics.store.StatisticsDataStore
 import kotlinx.coroutines.withTimeoutOrNull
 
+interface AtbInitializerListener {
+
+    /**
+     * This method will be called before initializing the ATB
+     */
+    suspend fun beforeAtbInit()
+
+    /**
+     * @return the timeout in milliseconds after which [beforeAtbInit]
+     * will be stopped
+     */
+    fun beforeAtbInitTimeoutMillis(): Long
+}
+
 class AtbInitializer(
     private val statisticsDataStore: StatisticsDataStore,
     private val statisticsUpdater: StatisticsUpdater,
-    private val appReferrerStateListener: AppInstallationReferrerStateListener
+    private val listeners: Set<AtbInitializerListener>
 ) {
```

Comment:
Creater `AtbInitializerListener` to break circular dependency with the main `app` module.
Also, the fact that the `AtbInitializer` depended on the `AppInstallationReferrerStateListener` was leaking the referrer logic into the `AtbInitializer`, which preferably should not happen.

The `AtbInitializer` now provides a listener that can be implemented by any other class in the app that is interested in executing some code before the `AtbInitializer` is initialized.
This way, the `AtbInitializer` does not need to know anything about these classes